index.bs

```diff
@@ -6438,6 +6445,90 @@ requests will be affected.
 
 </div>
 
+#### The network.setCacheMode Command ####  {#command-network-setCacheMode}
+
+The <dfn export for=commands>network.removeSetCacheMode</dfn> command overrides
+the network cache behaviour for certain requests.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      network.SetCacheMode = (
+        method: "network.setCacheMode",
+        params: network.SetCacheModeParameters
+      )
+
+      network.SetCacheModeParameters = {
+        mode: "no-store" / null,
+        ? contexts: [+browsingContext.BrowsingContext]
+      }
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl">
+      EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm>
+The <dfn export>WebDriver BiDi update request cache mode</dfn> steps given
+|request| are:
+
+1. Let |context| be null.
+
+1. If |request|'s [=request/window=] is an [=environment settings object=]:
+
+  1. Let |environment settings| be |request|'s [=request/window=]
+
+  1. If there is a [=/browsing context=] whose [=active window=] is |environment
+     settings|' [=environment settings object/global object=], set |context| to
+     the [=top-level browsing context=] for that context.
+
+1. If |context| is not null:
+
+  1. If [=cache mode override map=] [=map/contains=] |context|, set
+     |request|&apos;s [=request/cache mode=] to [=cache mode override
+     map=][|context|] and return.
+
+1. If [=default cache mode override=] is not null, set
+     |request|&apos;s [=request/cache mode=] to [=default cache mode override=].
+
+</div>
+
+<div algorithm="remote end steps for network.setCacheMode">
+The [=remote end steps=] given <var ignore>session</var> and |command parameters| are:
+
+1. Let |mode| be |command parameters|["<code>mode</code>"].
+
+1. If |command parameters| does not [=map/contain=] "<code>contexts</code>", set
+   the [=default cache mode override=] to |mode| and return [=success=] with
+   data null.
+
+1. Let |contexts| be an empty [=/list=].
+
+1. For each |context id| of |command parameters|["<code>contexts</code>"]:
+
+  1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+     with |context id|.
+
+  1. If |context| is not a [=top-level browsing context=], return [=error=]
+     with [=error code=] [=invalid argument=].
+
+  1. [=list/Append=] |context| to |contexts|.
+
+1. For each |context| in |contexts|:
```

Comment:
We run the loop twice. Should we combine steps 4 and 5?

Reply:
We can't. We first need to check that we have only valid context ids and then perform updates. Otherwise a contexts list like `[valid context, invalid context]` would update the state for `valid context` before throwing an error.

Reply:
Good point! That would be indeed an unwanted behavior.